File: 1_algorithmic_toolbox/week3/python/7_largest_concatenate.py
if __name__ == "__main__":
    from functools import cmp_to_key

    input()  # skip length of sequence input
    sequence = [i.strip() for i in input().split()]
    sequence.sort(
        key=cmp_to_key(
            lambda s1, s2: 1 if s2 + s1 > s1 + s2 else -1 if s2 + s1 < s1 + s2 else 0
        )
    )
    print("".join(sequence))

File: 1_algorithmic_toolbox/week3/rust/7_largest_concatenate.rs
fn main() {
    let mut buffer = String::new();
    for _ in 0..2 {
        std::io::stdin().read_line(&mut buffer).unwrap();
    }
    let mut input = buffer.split_whitespace();
    input.next(); // skip length of sequence input

    let mut sequence = input.collect::<Vec<_>>();
    sequence.sort_by(|s1, s2| (s2.to_string() + s1).cmp(&(s1.to_string() + s2)));

    println!("{}", sequence[..].join(""));
}
